etc/scripts/travis/upload_logs

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+
+var fs = require('fs');
+var amazons3 = require('awssum-amazon-s3');
+var path = require('path');
+var util = require('util');
+
+//////////////////////
+// HANDLE ARGUMENTS //
+//////////////////////
+
+var argv = require('yargs')
+    .usage('Usage: $0 -l <path of log file> -b <bucket name> [-r <region=eu-west-1>]')
+
+    .alias('h', 'help')
+    .describe('h', 'Show help information')
+
+    .alias('l', 'log')
+    .describe('l', 'The path to the log file that should be uploaded')
+    .default('l', 'tests.log.gz')
+
+    .alias('b', 'bucket')
+    .describe('b', 'The Amazon S3 bucket to which to upload the log file')
+    .default('b', 'fronteer-grasshopper')
+
+    .alias('r', 'region')
+    .describe('r', 'The Amazon region to which to upload the log file')
+    .default('r', 'eu-west-1')
+
+    .argv;
+
+// Display the help if requested
+if (argv.h) {
+    require('yargs').showHelp();
+    return process.exit(0);
+}
+
+// Ensure that we're on the master branch
+if (process.env['TRAVIS_REPO_SLUG'] !== 'fronteerio/grasshopper' || process.env['TRAVIS_BRANCH'] !== 'master' || process.env['TRAVIS_PULL_REQUEST'] == 'true') {
+    console.log('Only uploading logs for commits on fronteerio/grasshopper#master');
+    return;
+}
+
+
+// Get the argument values
+var bucketName = argv.b;
+var region = argv.r;
+var logPath = argv.l;
+
+// Stat the log file to a/ ensure that it exists and b/ get the size of the file
+fs.stat(logPath, function(err, stat) {
+    if (err) {
+        console.log('Failed to get the file information for the test log file');
```

Comment:
Not really, that would stringify the `err` object, it's a bit easier to debug if the full error object is logged.